### PR TITLE
samples: matter: add matter tag for samples

### PR DESCRIPTION
**Pull request by nordic-piks**
Extend list of paths used to check if matter
samples should be executed.

### CODE REVIEW

**Comment by nordic-piks** [comment]
The idea behind this change is to allow automatic detection of PRs not related to given code area.
In this case, there is defined tag (e.g. matter) and files/folder with source code related to that area.
This list of files/folders is then compared to files changed in PR. If particular PR does not change them,
tests related to specified tag should not be executed - will be excluded.

**Comment by nordic-piks** [comment]
Updated after evaluation of:
https://github.com/nrfconnect/sdk-nrf/blob/main/.github/test-spec.yml#L351-L375
https://github.com/nrfconnect/sdk-zephyr/blob/main/.github/test-spec.yml#L178-L191
https://github.com/nrfconnect/sdk-nrfxlib/blob/main/.github/test-spec.yml#L107-L119
https://github.com/nrfconnect/suit-generator/blob/main/.github/test-spec.yml#L4-L5

Please note that the content is similar and syntax is different - the root path is NCS, all dependencies - from west modules - are stated at one place, tests and samples should not be included here, they are handled separately (but they will be build if changed)

**Comment by nordic-piks** [comment]
DNM until NCS 2.7

**Comment by doublemis1** [comment]
@nordic-piks 
You can add `matter` to `applications/matter_weather_station` and `applications/matter_bridge`

**Comment by nordic-piks** [comment]
> @nordic-piks You can add `matter` tag to `applications/matter_weather_station` and `applications/matter_bridge`

Thanks, added.

**Review by tejlmand — APPROVED** [review]

**Comment by ArekBalysNordic** [comment]
If I understood the idea of this PR correctly, we have a snippet in the `snippets/matter-diagnostic-logs` directory that can be added to the tag list.

**Comment by nordic-piks** [comment]
> If I understood the idea of this PR correctly, we have a snippet in the `snippets/matter-diagnostic-logs` directory that can be added to the tag list.

Done, thanks.

**Review by nordic-segl — APPROVED** [review]

File: scripts/ci/tags.yaml [hunk 1]
```diff
@@ -106,3 +106,39 @@ wifi:
     - zephyr/subsys/net/l2/ethernet/
     - nrfxlib/nrf_wifi/
     - nrfxlib/crypto/
+
+matter:
+  files:
```
Comment (doublemis1) [inline scripts/ci/tags.yaml:111]: if this is a list of also depedency within ncs submodules we need to check if this is all. 
for sure I don't see modules/lib/matter

**Review by ArekBalysNordic — APPROVED** [review]

File: scripts/ci/tags.yaml (hunk 1 above)
Comment (nordic-piks) [reply scripts/ci/tags.yaml:111]: Added, thanks :)

**Comment by nordic-piks** [comment]
> ### Memory footprint analysis revealed the following potential issues
> _**sample.matter.template.release[nrf5340dk/nrf5340/cpuapp]**_: _**RAM**_ size _**increased by 628[B]**_ in comparison to the _**main[[80832d7](https://github.com/nrfconnect/sdk-nrf/commit/80832d7fd581a58dfe3dfaa509d1392d9ace4e4b)]**_ branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=sample&var-submodule=matter&var-scenario=template&var-testcase=release&var-arch=arm&var-platform=nrf5340dk/nrf5340/cpuapp) (cc: @kkasperczyk-no @ArekBalysNordic @markaj-nordic) _**sample.matter.template.release[nrf7002dk/nrf5340/cpuapp]**_: _**RAM**_ size _**increased by 628[B]**_ in comparison to the _**main[[80832d7](https://github.com/nrfconnect/sdk-nrf/commit/80832d7fd581a58dfe3dfaa509d1392d9ace4e4b)]**_ branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=sample&var-submodule=matter&var-scenario=template&var-testcase=release&var-arch=arm&var-platform=nrf7002dk/nrf5340/cpuapp) (cc: @kkasperczyk-no @ArekBalysNordic @markaj-nordic)
> 
> _Note: This message is automatically posted and updated by the CI (latest/sdk-nrf/PR-16171/6)_

This PR is not touching the code.
Instable measurements? @katgiadla ?

**Review by doublemis1 — APPROVED** [review]
LGTM

**Comment by nordic-piks** [comment]
> > ### Memory footprint analysis revealed the following potential issues
> > _**sample.matter.template.release[nrf5340dk/nrf5340/cpuapp]**_: _**RAM**_ size _**increased by 628[B]**_ in comparison to the _**main[[80832d7](https://github.com/nrfconnect/sdk-nrf/commit/80832d7fd581a58dfe3dfaa509d1392d9ace4e4b)]**_ branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=sample&var-submodule=matter&var-scenario=template&var-testcase=release&var-arch=arm&var-platform=nrf5340dk/nrf5340/cpuapp) (cc: @kkasperczyk-no @ArekBalysNordic @markaj-nordic) _**sample.matter.template.release[nrf7002dk/nrf5340/cpuapp]**_: _**RAM**_ size _**increased by 628[B]**_ in comparison to the _**main[[80832d7](https://github.com/nrfconnect/sdk-nrf/commit/80832d7fd581a58dfe3dfaa509d1392d9ace4e4b)]**_ branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=sample&var-submodule=matter&var-scenario=template&var-testcase=release&var-arch=arm&var-platform=nrf7002dk/nrf5340/cpuapp) (cc: @kkasperczyk-no @ArekBalysNordic @markaj-nordic)
> > _Note: This message is automatically posted and updated by the CI (latest/sdk-nrf/PR-16171/6)_
> 
> This PR is not touching the code. Instable measurements? @katgiadla ?

Discussed with @katgiadla, this is tool imperfection.

**Comment by nordic-piks** [comment]
Added also paths with tests.

**Pull request merged**